sources/networkmanager/transit_gateway_registration.go

```diff
@@ -0,0 +1,151 @@
+package networkmanager
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/networkmanager"
+	"github.com/aws/aws-sdk-go-v2/service/networkmanager/types"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+func transitGatewayRegistrationOutputMapper(_ context.Context, _ *networkmanager.Client, scope string, _ *networkmanager.GetTransitGatewayRegistrationsInput, output *networkmanager.GetTransitGatewayRegistrationsOutput) ([]*sdp.Item, error) {
+	items := make([]*sdp.Item, 0)
+
+	for _, r := range output.TransitGatewayRegistrations {
+		var err error
+		var attrs *sdp.ItemAttributes
+		attrs, err = sources.ToAttributesCase(r, "tags")
+
+		if err != nil {
+			return nil, &sdp.QueryError{
+				ErrorType:   sdp.QueryError_OTHER,
+				ErrorString: err.Error(),
+				Scope:       scope,
+			}
+		}
+
+		attrs.Set("globalNetworkWithTransitGatewayARN", fmt.Sprintf("%s|%s", *r.GlobalNetworkId, *r.TransitGatewayArn))
+
+		item := sdp.Item{
+			Type:            "networkmanager-transit-gateway-registration",
+			UniqueAttribute: "globalNetworkWithTransitGatewayARN",
+			Scope:           scope,
+			Attributes:      attrs,
+			LinkedItemQueries: []*sdp.LinkedItemQuery{
+				{
+					Query: &sdp.Query{
+						// +overmind:link networkmanager-global-network
+						Type:   "networkmanager-global-network",
+						Method: sdp.QueryMethod_GET,
+						Query:  *r.GlobalNetworkId,
+						Scope:  scope,
+					},
+					BlastPropagation: &sdp.BlastPropagation{
+						In:  true,
+						Out: false,
+					},
+				},
+			},
+		}
+
+		// TODO: add support for ec2-transit-gateway
+		// ARN example: "arn:aws:ec2:us-west-2:123456789012:transit-gateway/tgw-1234"
+		if r.TransitGatewayArn != nil {
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					// +overmind:link ec2-transit-gateway
+					Type:   "ec2-transit-gateway",
+					Method: sdp.QueryMethod_SEARCH,
+					Query:  *r.TransitGatewayArn,
+					Scope:  scope,
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					In:  true,
+					Out: false,
+				},
+			})
+		}
+
+		if r.State != nil {
+			switch r.State.Code {
+			case types.TransitGatewayRegistrationStatePending:
+				item.Health = sdp.Health_HEALTH_PENDING.Enum()
+			case types.TransitGatewayRegistrationStateAvailable:
+				item.Health = sdp.Health_HEALTH_OK.Enum()
+			case types.TransitGatewayRegistrationStateDeleting:
+				item.Health = sdp.Health_HEALTH_PENDING.Enum()
+			case types.TransitGatewayRegistrationStateDeleted:
```

Comment:
As with the other one I'd say this should probably just be `nil`, the logic being that if it has been deleted it doesn't have health